server/dumper.js

```diff
@@ -279,19 +280,25 @@ Dumper.prototype.exprFor_ = function(value, ref, callable, funcName) {
 
   // Return existing reference to object (if already created).
   var objDumper = this.getObjectDumper_(value);
-  var prev;
-  if (objDumper.ref) prev = this.exprForSelector_(objDumper.getSelector());
+  var selector;  // Existing selector for value, if any.
+  if (objDumper.proto !== undefined && objDumper.ref) {
+    selector = objDumper.getSelector();
+  }
   if (ref) objDumper.updateRef(ref);  // Safe new ref if specified.
-  if (prev) return prev;
+  if (selector) return this.exprForSelector_(selector);
 
  // Object not yet referenced.  Is it a builtin?
   var key = intrp2.builtins.getKey(value);
   if (key) {
     var quoted = code.quote(key);
     return callable ? '(new ' + quoted + ')' : 'new ' + quoted;
   }
-  // New object.  Create and save referece for later use.
-  if (!ref) throw Error('Refusing to create non-referable object');
+
+  // Seems to be a new object.  Check it really doesn't exist already
+  // and that it will be referenceable.
+  if (objDumper.proto !== undefined) throw new Error('object already exists');
```

Comment:
"throw new Error" here, "throw Error" below.  Pick one.

Reply:
Style guide says to use "new".  I think